unit/goto-programs/goto_program_generics/generic_parameters_test.cpp

```diff
@@ -486,3 +487,124 @@ SCENARIO(
     }
   }
 }
+
+SCENARIO(
+  "Ignore generic parameters in fields and methods with incomplete and "
+  "non-generic types",
+  "[core][goto_program_generics][generic_parameters_test]")
+{
+  GIVEN(
+    "A class with a generic field pointing to a class with unsupported "
+    "signature (thus not marked as generic)")
+  {
+    const symbol_tablet &symbol_table = load_java_class(
+      "GenericFieldUnsupported",
+      "./goto-programs/goto_program_generics",
+      "GenericFieldUnsupported.foo");
+
+    THEN("The struct for UnsupportedWrapper2 is complete and non-generic")
+    {
+      const std::string superclass_name = "java::UnsupportedWrapper2";
+      REQUIRE(symbol_table.has_symbol(superclass_name));
+
+      const symbolt &superclass_symbol =
+        symbol_table.lookup_ref(superclass_name);
+      require_type::require_java_non_generic_class(superclass_symbol.type);
+    }
+
+    WHEN("The method input argument is created in the entry point function")
+    {
+      // For an explanation of this part, look at the comments for the similar
+      // parts of the previous tests.
```

Comment:
While I realise that this has become something of a boiler plate comment, copy and pasted from other test cases, I actually think there would be more value in copy and pasting the *actual* comment, rather than copy and pasting the reference back to previous tests...